### ISSUE TITLE
[board] Hardcoded values

### ISSUE THREAD

**Issue by heyapricot** [rvvergara/tictactoe#9]
1. On your `Board` class, you define an array with all the winning combinations. This is not bad `per se` but I don't like it since it is similar to the use of [magic numbers](https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad).

A better solution in my opinion is to implement a method that checks if all the elements in a row, column or diagonal are the same. This method is extensible since it will work with matrix of size `N`. Not only on a 3 x 3 matrix

**Comment by rvvergara** [comment]
Hi @heyapricot, thanks! You mentioned it's not bad per se. So what are the disadvantage performance wise or even maintenance wise if we keep the winning combinations as a constant with hardcoded numbers? 

I ask this because the way we implemented the winning combination check is similar to how Freecodecamp implemented their [tictactoe](https://www.youtube.com/watch?v=P2TcQ3h0ipQ) game (from which we derived our algorithm for the AI, though they didn't use OOP in building this game).
